--- rawstudio-2.0/librawstudio/Makefile.am.linking	2012-04-10 16:15:48.498525922 +0200
+++ rawstudio-2.0/librawstudio/Makefile.am	2012-04-10 16:16:40.018525959 +0200
@@ -112,7 +112,7 @@ librawstudio_2_0_la_SOURCES = \
 	rs-gui-functions.c rs-gui-functions.h \
 	rs-stock.c rs-stock.h
 
-librawstudio_2_0_la_LIBADD = @PACKAGE_LIBS@ @LIBJPEG@ @LIBTIFF@ $(INTLLIBS)
+librawstudio_2_0_la_LIBADD = @PACKAGE_LIBS@ @LIBJPEG@ @LIBTIFF@ $(INTLLIBS) -lgmodule-2.0
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = rawstudio-2.0.pc
